pyClient/README.md

```diff
@@ -45,11 +55,192 @@ and fails to fetch it (or fails to fetch the right version).
+tools assume it can be found in specific files (with default file names,
+overridable on the command line).
+
+The set of files required for by zeth for a single user is described below.  We
+recommend creating a directory for each user, containing these files.  In this
+way, it is very easy to setup one or more conceptual "users" and invoke `zeth`
+operations on behalf of each of them to experiment with the system.
+
+- `eth-address` specifies an eth address from which to fund transactions.  When
+  running the testnet (as described in the [top-level README](../README.md)),
+  addresses are created at startup and written to the console.  One of these can
+  be copy-pasted into this file.
+- `zeth-instance.json` contains the address and interface for an instance of the
+  zeth contracts.  This file is created by the deployment step below and should
+  be distributed to each client.
+- `zeth-key.json` and `zeth-key.json.pub` hold the secret and public Zeth keys
+  for the client.  These can be generated with the `zeth key` command.
+  `zeth-key.json.pub` represents the public address of the client.  This can be
+  shared freely, allowing other users to privately transfer funds to this
+  client.  The secret `zeth-key.json` should **not** be shared.
+
+## Deployment
+
+Deployment compiles and deploys the contracts and initializes them with
+appropriate data to create a new empty instance of the Zeth mixer.  It requires
+only an `eth-address` file mentioned above, where the address has sufficient
+funds.
+
+```console
+# Create a clean directory for the deployer
+(env)$ mkdir deployer
+(env)$ cd deployer
+
+# Specify an eth-address file for an (unlocked) Ethereum account
+(env)$ echo 0x.... > eth-address
+
+# Compile and deploy
+(env)$ zeth deploy
+
+# Share the instance file with all clients
+$ cp zeth-instance.json <destination>
+```
+
+## User setup
+
+To set up her client, Alice must setup all client files mentioned above:
+```console
+# Create a clean client directory
+$ mkdir alice
+$ cd alice
+
+# Specify an eth-address file for an (unlocked) Ethereum account
+$ echo 0x.... > eth-address
+
+# Copy the instance file (received from the deployer)
+$ cp <shared-instance-file> zeth-instance.json
+
+# Generate new secret (zeth-key.json)and public key (zeth-key.json.pub)
+$ zeth key
+
+# Share the public address with other users
+$ cp zeth-key.json.pub <destination>
+```
+
+With these files in place, `zeth` commands invoked from inside this directory
+can perform actions on behalf of Alice.  We call this Alice's *client directory*
+below, and assume that all commands are executed in a directory with these
+files.
+
+## Depositing funds
+
+Alice can deposit funds from the address in `eth-address` into the Zeth
+instance, as a private Zeth note, spendable with her secret key `zeth-key.json`:
+
+```console
+# Deposit 10 ether, creating a Zeth note owned by Alice
+(env)$ zeth deposit 10
```

Comment:
May be worth adding a flag to be able to use ERC tokens @dtebbs 

Something like:
```console
zeth deposit 10 --token <erc-token-contract-address>
```

If no flag is given, it defaults to ETH

Reply:
I've added a `--token-address` flag to the `deploy` command.  The token address gets saved alongn with the zeth mixer instance data and used in the `mix` operations.  I've also added a `zeth token_approve` command for dealing with token.
Finally, a `deploy_test_token` command (in `test_commands`, not part of the main cli interface) allows the minimal token implementation to be deployed for demos / testing.